src/main/java/com/microsoft/azure/search/samples/client/SearchIndexClient.java

```diff
@@ -1,137 +1,109 @@
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLEncoder;
 import java.util.ArrayList;
-import java.util.Collection;
+import java.util.List;
 import java.util.Map;
 
 public class SearchIndexClient {
     private static final String API_VERSION = "2016-09-01";
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().registerModule(new Jdk8Module()).setDateFormat(
+            new ISO8601DateFormat());
 
     private final String serviceName;
     private final String indexName;
     private final String apiKey;
 
-    static {
-        OBJECT_MAPPER.setDateFormat(new ISO8601DateFormat());
-    }
-
     public SearchIndexClient(String serviceName, String indexName, String apiKey) {
         this.serviceName = serviceName;
         this.indexName = indexName;
         this.apiKey = apiKey;
     }
 
-    public boolean exists() throws IOException {
+    public boolean isIndexExists() throws IOException {
         HttpURLConnection connection = httpRequest(buildIndexDefinitionUrl(), "GET");
         int response = connection.getResponseCode();
-        if (response == 404) {
+        if (response == HttpURLConnection.HTTP_NOT_FOUND) {
             return false;
         }
         throwOnHttpError(connection);
         return true;
     }
 
-    public void create(IndexDefinition indexDefinition) throws IOException {
-        if (indexDefinition.getName() == null) {
-            indexDefinition.setName(this.indexName);
-        }
+    public void createIndex(IndexDefinition indexDefinition) throws IOException {
         HttpURLConnection connection = httpRequest(buildIndexListUrl(), "POST");
         connection.setDoOutput(true);
-        OBJECT_MAPPER.writeValue(connection.getOutputStream(), indexDefinition);
+        OBJECT_MAPPER.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY).writeValue(
+                connection.getOutputStream(), indexDefinition);
         throwOnHttpError(connection);
     }
 
-    public void createOrUpdate(IndexDefinition indexDefinition) throws IOException {
-        if (indexDefinition.getName() == null) {
-            indexDefinition.setName(this.indexName);
-        }
-        HttpURLConnection connection = httpRequest(buildIndexDefinitionUrl(), "PUT");
-        connection.setDoOutput(true);
-        OBJECT_MAPPER.writeValue(connection.getOutputStream(), indexDefinition);
-        throwOnHttpError(connection);
-    }
-
-    public boolean delete() throws IOException {
+    public void deleteIndexIfExists() throws IOException {
         HttpURLConnection connection = httpRequest(buildIndexDefinitionUrl(), "DELETE");
-        int response = connection.getResponseCode();
-        if (response == 404) {
-            return false;
-        }
         throwOnHttpError(connection);
-        return true;
```

Comment:
It can be useful to know whether the index was actually deleted or didn't exist; Why remove this functionality?

Also, the `IfExists` suffix suggests that the operation will fail if the index doesn't exist, which is not true.